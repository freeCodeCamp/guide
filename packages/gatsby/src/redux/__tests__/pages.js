const reducer = require(`../reducers/pages`)
const { actions } = require(`../actions`)

describe(`Add pages`, () => {
  it(`allows you to add pages`, () => {
    const action = actions.createPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
    })
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add pages with context`, () => {
    const action = actions.createPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
      context: {
        id: 123,
      },
    })
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add multiple pages`, () => {
    const action = actions.createPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
    })
    const action2 = actions.createPage({
      path: `/hi/pizza/`,
      component: `/whatever/index.js`,
    })
    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(2)
  })

  it(`allows you to update existing pages (based on path)`, () => {
    const action = actions.createPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
    })

    // Change the component
    const action2 = actions.createPage({
      path: `/hi/`,
      component: `/whatever2/index.js`,
    })

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(1)
  })

  it(`allows you to delete paths`, () => {
    const action = actions.createPage({
      path: `/hi/`,
      component: `/whatever/index.js`,
    })
    const action2 = actions.deletePage({ path: `/hi/` })

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(0)
  })
})
