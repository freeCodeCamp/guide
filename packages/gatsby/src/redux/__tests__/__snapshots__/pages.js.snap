// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add pages allows you to add multiple pages 1`] = `
Array [
  Object {
    "component": "/whatever/index.js",
    "componentChunkName": "page-component---whatever-index-js",
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
  },
  Object {
    "component": "/whatever/index.js",
    "componentChunkName": "page-component---whatever-index-js",
    "context": Object {},
    "internalComponentName": "ComponentHiPizza",
    "jsonName": "hi-pizza.json",
    "path": "/hi/pizza/",
  },
]
`;

exports[`Add pages allows you to add pages 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "page-component---whatever-index-js",
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
  },
  "plugin": "",
  "traceId": undefined,
  "type": "CREATE_PAGE",
}
`;

exports[`Add pages allows you to add pages 2`] = `
Array [
  Object {
    "component": "/whatever/index.js",
    "componentChunkName": "page-component---whatever-index-js",
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
  },
]
`;

exports[`Add pages allows you to add pages with context 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "page-component---whatever-index-js",
    "context": Object {
      "id": 123,
    },
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
  },
  "plugin": "",
  "traceId": undefined,
  "type": "CREATE_PAGE",
}
`;

exports[`Add pages allows you to add pages with context 2`] = `
Array [
  Object {
    "component": "/whatever/index.js",
    "componentChunkName": "page-component---whatever-index-js",
    "context": Object {
      "id": 123,
    },
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
  },
]
`;

exports[`Add pages allows you to delete paths 1`] = `Array []`;

exports[`Add pages allows you to update existing pages (based on path) 1`] = `
Array [
  Object {
    "component": "/whatever2/index.js",
    "componentChunkName": "page-component---whatever-2-index-js",
    "context": Object {},
    "internalComponentName": "ComponentHi",
    "jsonName": "hi.json",
    "path": "/hi/",
  },
]
`;
